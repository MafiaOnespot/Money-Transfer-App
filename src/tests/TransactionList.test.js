import React from "react";
import renderer from "react-test-renderer";
import TransactionList from "../components/TransactionList";

const transactions = [
  {
    isModeSend: true,
    executed: true,
    selectedName: "Barshan",
    amount: 10,
    newBalance: 990,
    date: "Feb 7",
    note: "lunch"
  },
  {
    isModeSend: false,
    executed: true,
    selectedName: "Devdulal",
    amount: 10,
    newBalance: 990,
    date: "Feb 2",
    note: "beer"
  },
  {
    isModeSend: true,
    executed: true,
    selectedName: "Shreya",
    amount: 10,
    newBalance: 990,
    date: "Jan 20",
    note: "cab share"
  },
  {
    isModeSend: false,
    executed: true,
    selectedName: "Manasij",
    amount: 10,
    newBalance: 990,
    date: "Jan 18",
    note: "cinema"
  }
];

it("renders correctly", () => {
  const tree = renderer
    .create(<TransactionList transactions={transactions} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
