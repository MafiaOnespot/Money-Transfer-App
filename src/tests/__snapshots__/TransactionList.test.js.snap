// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="transactions"
>
  <h2>
    History
  </h2>
  <div
    className="transaction"
  >
    <div
      className="transaction__left__left"
    >
      <div
        className="transaction__left__message"
      >
        Sent
         ₹
        10
         
        to
         
        Barshan
      </div>
      <div
        className="transaction__left__note"
      >
        lunch
      </div>
    </div>
    <div
      className="transaction__right"
    >
      <div
        className="transaction__right__date"
      >
        Feb 7
      </div>
      <div
        className="transaction__right__balance"
      >
         
        Balance: ₹
        990
      </div>
    </div>
  </div>
  <div
    className="transaction"
  >
    <div
      className="transaction__left__left"
    >
      <div
        className="transaction__left__message"
      >
        Received
        ₹
        10
         
        from
         Devdulal
        
      </div>
      
    </div>
    <div
      className="transaction__right"
    >
      <div
        className="transaction__right__date"
      >
        Feb 2
      </div>
      <div
        className="transaction__right__balance"
      >
         
        Balance: ₹
        1000
      </div>
    </div>
  </div>
  <div
    className="transaction"
  >
    <div
      className="transaction__left__left"
    >
      <div
        className="transaction__left__message"
      >
        Sent
         ₹
        10
         
        to
         
        Shreya
      </div>
      <div
        className="transaction__left__note"
      >
        cab share
      </div>
    </div>
    <div
      className="transaction__right"
    >
      <div
        className="transaction__right__date"
      >
        Jan 20
      </div>
      <div
        className="transaction__right__balance"
      >
         
        Balance: ₹
        990
      </div>
    </div>
  </div>
  <div
    className="transaction"
  >
    <div
      className="transaction__left__left"
    >
      <div
        className="transaction__left__message"
      >
        Received
         ₹
        10
         
        from
         
        Manasij
      </div>
      <div
        className="transaction__left__note"
      >
        cinema
      </div>
    </div>
    <div
      className="transaction__right"
    >
      <div
        className="transaction__right__date"
      >
        Jan 18
      </div>
      <div
        className="transaction__right__balance"
      >
         
        Balance: ₹
        1000
      
      </div>
    </div>
  </div>
</div>
`;
